src/storage/core.rs

```diff
@@ -303,10 +304,42 @@ impl<K: Key> Storage<K> {
         }
     }
 
-    async fn get_any_data(safe: &Safe, key: &[u8], meta: Option<&Meta>) -> Result<Vec<u8>> {
+    async fn get_data_last(safe: &Safe, key: &[u8], meta: Option<&Meta>) -> Result<Vec<u8>> {
+        let blobs = safe.blobs.read().await;
+        let blobs_stream: FuturesOrdered<_> = blobs
+            .iter()
+            .enumerate()
+            .map(|(i, blob)| async move {
+                if blob.check_filters_non_blocking(key).await {
+                    Some(i)
+                } else {
+                    None
+                }
+            })
+            .collect();
+        let possible_blobs: Vec<_> = blobs_stream.filter_map(|e| e).collect().await;
```

Comment:
Why you didn't use `filter_map` on L312?

Reply:
Because closure in that `map` returns `future` (async block), but `filter_map` expects `Option`.
Resolve conversation if the answer is ok.